cheroot/server.py

```diff
@@ -1186,12 +1189,25 @@ def ensure_headers_sent(self):
 
     def write(self, chunk):
         """Write unbuffered data to the client."""
-        if self.chunked_write and chunk:
-            chunk_size_hex = hex(len(chunk))[2:].encode('ascii')
-            buf = [chunk_size_hex, CRLF, chunk, CRLF]
-            self.conn.wfile.write(EMPTY.join(buf))
-        else:
-            self.conn.wfile.write(chunk)
+        try:
+            if self.chunked_write and chunk:
+                chunk_size_hex = hex(len(chunk))[2:].encode('ascii')
+                buf = [chunk_size_hex, CRLF, chunk, CRLF]
+                self.conn.wfile.write(EMPTY.join(buf))
+            else:
+                self.conn.wfile.write(chunk)
+        except (SysCallError, ConnectionError, OSError) as e:
+            error_code = e.errno if isinstance(e, OSError) else e.args[0]
```

Comment:
There's a number of places where this is repeated. It's worth having a helper function, I think.